shell/platform/darwin/ios/framework/Source/FlutterEngineGroup.mm

```diff
@@ -0,0 +1,60 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/ios/framework/Headers/FlutterEngineGroup.h"
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterEngine_Internal.h"
+
+@interface FlutterEngineGroup ()
+@property(nonatomic, copy) NSString* name;
+@property(nonatomic, strong) NSMutableArray<NSValue*>* engines;
+@property(nonatomic, strong) FlutterDartProject* project;
+@end
+
+@implementation FlutterEngineGroup {
+  int _engineCount;
+}
+
+- (instancetype)initWithName:(NSString*)name project:(nullable FlutterDartProject*)project {
+  self = [super init];
+  if (self) {
+    self.name = name;
+    self.engines = [[NSMutableArray<NSValue*> alloc] init];
+    self.project = project;
+  }
+  return self;
+}
+
+- (void)dealloc {
+  NSNotificationCenter* center = [NSNotificationCenter defaultCenter];
+  [center removeObserver:self];
+  [_name release];
+  [_engines release];
+  [super dealloc];
+}
+
+- (FlutterEngine*)makeEngineWithEntrypoint:(nullable NSString*)entrypoint {
```

Comment:
This is really makeAndRunEngine right? Make the other constructor permutations such as the basic run main in main.dart? And also for specifying a .dart file and entrypoint?

Reply:
addressed in "xiao feedback 1" -- Added clarification in the docstring.